src/routes/applications.ts

```diff
@@ -15,32 +13,31 @@ import { RequestHandler, Request } from 'express';
 
 const { UnauthorizedError, api } = sbvrUtils;
 
-const getApp = (req: Request): Promise<AnyObject> =>
-	api.resin
-		.get({
-			resource: 'application',
-			id: req.param('appId'),
-			passthrough: { req },
-			options: {
-				$select: ['id', 'app_name'],
-				$expand: {
-					is_for__device_type: {
-						$select: ['slug'],
-					},
+const getApp = async (req: Request): Promise<AnyObject> => {
+	const app = (await api.resin.get({
+		resource: 'application',
+		id: req.param('appId'),
+		passthrough: { req },
+		options: {
+			$select: ['id', 'app_name'],
+			$expand: {
+				is_for__device_type: {
+					$select: ['slug'],
 				},
 			},
-		})
-		.then((app: AnyObject) => {
-			// Check that the current user has access to this application.
-			if (app == null) {
-				throw new UnauthorizedError(
-					'You do not have permission to access this application',
-				);
-			}
-			return app;
-		});
+		},
+	})) as AnyObject;
```

Comment:
We really need to sort the types, the need to cast every single query to AnyObject is unreasonable.

Reply:
It's actually the same as before in terms of casting, just the cast was non-obvious before because we don't enforce `strictFunctionTypes` and so can do it via `(x: cast) =>` in function params, and did because the true cast in promise world gets much messier than this much quicker. And ideally we will switch to enforcing `strictFunctionTypes` because the ability to implicitly cast is a potential source of non-obvious errors.
That said we do need a way to handle better types in pinejs-client, it's just the `'/$count'` stuff that makes it really hard to to type it in a way it narrows to the single return type it can be

Reply:
The problematic case is just “.get” and we don’t *have* to overload it. We could have “get” that requires an “id” and fetches zero or one results, “find<T>” that takes no “id” and requires a “$filter” and returns a “T[]”, and finally a “count” that does a count based on given “$filter” and solves the problematic case you mention. Let’s discuss this some more in FD